frame/support/test/tests/pallet.rs

```diff
@@ -399,13 +399,18 @@ pub mod pallet {
 	pub type Unbounded<T> = StorageValue<Value = Vec<u8>>;
 
 	#[pallet::genesis_config]
-	#[derive(Default)]
-	pub struct GenesisConfig {
+	#[derive(frame_support::DefaultNoBound)]
+	pub struct GenesisConfig<T: Config>
+	where
+		T::AccountId: From<SomeType1> + SomeAssociation1 + From<SomeType4>,
+	{
+		#[serde(skip)]
+		_config: sp_std::marker::PhantomData<T>,
```

Comment:
Is annoying that serde_json doesn't skip `PhantomData` and writes `null`.
IIRC scale codec defines the serdialization as a no-op

Reply:
Yep, I did not find the way how to do it more elegantly. We could all phantom into macro, but it is tricky also, and probably not a good way to do.